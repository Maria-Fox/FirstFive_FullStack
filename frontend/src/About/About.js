import React from "react";
import { Card, CardTitle, CardText } from "reactstrap";

const About = () => {
  return (
    <Card className="container bg-dark text-white">
      <CardTitle tag="h1" className="text-center  p-2 mt-3">About FirstFive</CardTitle>


      <CardText>FirstFive is a group project collaboration app with a "dating-app" aspect for added fun!</CardText>

      <CardText>Here, users can review and match projects. Once a user has matched a project they can see a list of all other users who matched the project or are active project members. From there, matched users can begin messaging one another and move forward with the project.
      </CardText>

      <h3 className="text-center m-3 mt-5">Why was FirstFive created?</h3>
      <CardText>FirstFive was created to support early career professionals in gaining valuable collaboration experience.</CardText>

      <h4 className="text-center m-3 mt-5">Why the name FirstFive?</h4>
      <CardText>
        The typical computer science degree takes 4-5 years. Bootcamps vary. However, a common theme among both graduates is the idea that they have few projects to demonstrate their skills, or they are looking for group proejcts to strengthen their skills.
      </CardText>

      <CardText>
        FirstFive is intended for those with 0 - 5 years of experience in an indsutry. Instead of completing a degree or bootcamp with little or no projects here users can slowly build up their resume while aquiring necessary collaboration skills.
      </CardText>

      <CardText>
        Users of all skill levels and years of experience are welcome.
      </CardText>
    </Card>
  );
};

export default About;