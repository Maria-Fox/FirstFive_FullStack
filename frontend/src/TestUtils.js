// Used to create sample test user data & test UserContect provider..

import React from "react";
import UserContext from "./UserComponents/UserContext";

const authUser = {
  username: "softwareDev1"
};

// sample / data
const matchedProjectIds = [1, 2, 3, 4];

const userWithMatchState = { username: "softwareDev1", matchedProjectIds: [] };

// Creating a context provider and passing in the test user, matching ids. Passing in chidlren.
const UserProvider =
  ({ children, userWithMatchState }) => {
    <UserContext.Provider value={userWithMatchState}>
      {children}
    </UserContext.Provider>
  };

export { UserProvider, authUser, userWithMatchState };