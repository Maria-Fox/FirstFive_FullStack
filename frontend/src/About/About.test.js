import React from "react";
import { render } from "@testing-library/react";
import { UserProvider } from "../TestUtils";
import About from "./About";

test("Smoke - About renders", function () {
  render(<About />);
});

test("About renders for valid user", function () {

  let { asFragment } = render(
    <UserProvider>
      <About />
    </UserProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("About does NOT render for invalid user", function () {

  let { asFragment } = render(
    <About />
  );

  expect(asFragment()).not.toMatchSnapshot();
});