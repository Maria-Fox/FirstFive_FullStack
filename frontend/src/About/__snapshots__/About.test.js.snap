// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders for valid user 1`] = `<DocumentFragment />`;
